### PR TITLE
SAK-45601: Modify "ifs" processing for XAPI events and pool usage instead of threads

### PR DESCRIPTION
**Pull request by victorGomollon**
https://jira.sakaiproject.org/browse/SAK-45601

### CODE REVIEW

**Review by ern — DISMISSED** [review]
Hi Victor this is a decent improvement. It would be ideal if there were unit tests around this functionality. I would also put the statements in queue this tends to work well when dealing with threads as it provides good handoff from a synchronous thread to asynchronous. When working with threads it's ideal to have an option where it can execute synchronously and asynchronous this is generally helpful for finding bugs.

**Comment by victorGomollon** [comment]
> 
> 
> Hi Victor this is a decent improvement. It would be ideal if there were unit tests around this functionality. I would also put the statements in queue this tends to work well when dealing with threads as it provides good handoff from a synchronous thread to asynchronous. When working with threads it's ideal to have an option where it can execute synchronously and asynchronous this is generally helpful for finding bugs.

Hi Earle,
What are you thinking about? I mean do you have some example of this synchronous vs asynchornous behavior?
thanks for comments.

**Comment by victorGomollon** [comment]
I see that there are no unit tests that can be adapted to test this, we have to mock up a lot of classes, so creating the tests is going to take a lot of time. Do you have any idea about what to do to speed up the process and make the creation of these tests less time consuming?

File: kernel/api/src/main/java/org/sakaiproject/event/api/TincanapiEvent.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+import lombok.Data;
+
+@Entity
+@Table( name = "TINCANAPI_EVENT")
+@NamedQuery( name = "getAllTincanapiEvent",
+			 query = "from TincanapiEvent")
+
+@Data
+public class TincanapiEvent {
```
Comment (ern) [inline kernel/api/src/main/java/org/sakaiproject/event/api/TincanapiEvent.java:17]: Is a table really needed, could a queue be sufficient?

Comment (SedueRey) [reply kernel/api/src/main/java/org/sakaiproject/event/api/TincanapiEvent.java:17]: In this case, the Table is used to save the different type of event, not the event itself. It's an easy way to add new ones.
The XApi event is not stored, it's just sent.
Thanks

**Comment by SedueRey** [comment]
@ern Unit test has been made. With the other comment we think this should be ok.

**Review by juanjmerono — APPROVED** [review]

**Pull request merged**